### PR TITLE
fix licencse check

### PR DESCRIPTION
**Pull request by the-other-tim-brown**
## *Important Read*
- *Please ensure the GitHub issue is mentioned at the beginning of the PR*

## What is the purpose of the pull request

*(For example: This pull request implements the sync for delta format.)*

## Brief change log

*(for example:)*
- *Fixed JSON parsing error when persisting state*
- *Added unit tests for schema evolution*

## Verify this pull request

*(Please pick either of the following options)*

This pull request is a trivial rework / code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

*(example:)*

- *Added integration tests for end-to-end.*
- *Added HudiSchemaEvolutionTest to verify the change.*
- *Manually verified the change by running a job locally.*

### CODE REVIEW

**Comment by the-other-tim-brown** [comment]
@wuchunfu can you help me to get the license check passing?

**Comment by charlesy6** [comment]
Having both `spotless check` and `skywalking-eys` in the same project is not a good practice, because their behaviors are not the same, what do you think?  @the-other-tim-brown @sagarlakshmipathy

**Comment by the-other-tim-brown** [comment]
@charlesy6 the intention was that the `skywalking` check would find anything not covered by the `spotless` plugin since the spotless plugin requires you to define rules for different file types not covered by default. If there is something that better fits that need, let me know.

**Comment by daragu** [comment]
@the-other-tim-brown, I agree with @charlesy6's point that there is no need to introduce the `skywalking` into xtable project.
and it is also not used in projects such as hudi, paimon, flink...

We can use `mvn validate` command to execute check commod on plugins such as `spotless`, `enforcer`.

E.g

```
name: CI mvn validate

concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true

on:
  push:
    branches:
      - "main"

  pull_request:
    branches:
      - "main"

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4

      - uses: actions/setup-java@v4
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: validate plugin
        run: mvn validate
```

**Comment by the-other-tim-brown** [comment]
> @the-other-tim-brown, I agree with @charlesy6's point that there is no need to introduce the `skywalking` into xtable project. and it is also not used in projects such as hudi, paimon, flink...
> 
> We can use `mvn validate` or `mvn spotless:check` command to execute check commod on plugins such as `spotless`, `enforcer`.
> 
> E.g
> 
> ```
> name: CI mvn validate
> 
> concurrency:
>   group: ${{ github.workflow }}-${{ github.ref }}
>   cancel-in-progress: true
> 
> on:
>   push:
>     branches:
>       - "main"
> 
>   pull_request:
>     branches:
>       - "main"
> 
> jobs:
>   build:
>     runs-on: ubuntu-latest
>     steps:
>       - uses: actions/checkout@v4
> 
>       - uses: actions/setup-java@v4
>         with:
>           java-version: '11'
>           distribution: 'temurin'
> 
>       - name: validate plugin
>         run: mvn validate
> ```

@daragu  how do we catch new file types being added without licenses? That is not currently possible with spotless. `mvn validate` is just part of the Maven lifecycle, you will run that as part of `install` https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html

**Comment by daragu** [comment]
thanks @the-other-tim-brown,  let me think again.

File: .licenserc.yaml [hunk 1]
```diff
@@ -43,4 +43,8 @@ header:
     - '**/LICENSE'
     - '**/NOTICE'
     - '**/.gitkeep'
+    - 'demo/data/**'
```
Comment (daragu) [inline .licenserc.yaml:46]: Hi, we no longer need this file `.licenserc.yaml`.

File: xtable-core/src/test/resources/schemas/basic_schema.avsc [hunk 2]
```diff
@@ -1,3 +1,20 @@
+/*
```
Comment (daragu) [inline xtable-core/src/test/resources/schemas/basic_schema.avsc:1]: The main branch already updated the license header.

**Comment by daragu** [comment]
hi @the-other-tim-brown, when a new type of file is added to the repo, it will be detected by the maven `rat` plugin automatically. it seems that we can close this pr now.

**Pull request closed without merging**